pickhardtpayments/Payment.py

```diff
@@ -160,15 +160,7 @@ def filter_attempts(self, flag: AttemptStatus) -> list[Attempt]:
         :return: A list of successful Attempts of this Payment, which could be settled.
         :rtype: list[Attempt]
         """
-        filtered_attempts = []
-        try:
-            for attempt in self._attempts:
-                if attempt.status.value == flag.value:
-                    filtered_attempts.append(attempt)
-            return filtered_attempts
-        except ValueError:
-            logging.warning("ValueError in Payment.filtered_attempts")
-        return []
+        return [attempt for attempt in self._attempts if attempt.status.value == flag.value]
```

Comment:
Haha! That is not what I meant by creating a generstor using `yield`. But it is great to see how python can reduce "trivial" code blocks. (: